### ISSUE TITLE
Services Menu / Name Organization

### ISSUE THREAD

**Issue by jeffg2k** [opnsense/core#2355]
In the services menu, I'd like to make some suggestions.  I don't mind coding these myself, but I'm not aware of the background on why these choices were made and if my suggestions would be accepted.  So if they sound like good ideas, I can try to implement them.

1. The menu name of "Dnsmasq DNS" and "Unbound DNS" are application names that many people may not know (I didn't).  Like the other services that use the general name, I think these would be better labeled "DNS Forwarder" and "DNS Resolver".  Those are common terms that most should know and describes what the services do, where application names don't.
2. DHCPv4, DHCPv6 - why two menu items?  How about one menu item "DHCP", where the pages for Relay & Leases have two tabs, one for IPv4 and another for IPv6.
3. Dnsmasq, OpenDNS, Unbound DNS - I"m not sure how to best deal with this, but it seems there should just be a "DNS" menu with various DNS services below it.  Not sure if the menu supports 4 levels, if DNS should be a top level with three subs, or if maybe OpenDNS should be someplace else.  But it's just odd to me that it has three DNS services that are not grouped.

**Comment by fichtner** [comment]
Hi @jeffg2k,

1. No because people used to talk a lot about the confusion of "Forwarder" and "Resolver" so we changed it. It's a general policy to try to push the actual software name into the menu for services like done in most plugins.

2. Firstly depth issues in the menu (we only try to keep it to 3 levels). Secondly the whole DHCP model is broken down into one IPv4 and IPv6 instances in the code so the menu follows it since the code was added. But open to suggestions that avoid the use of "v4" and "v6" more than once in the submenu.

3. Three levels. Easier pluginification in case a DNS service goes away. Clustering the services into categories is viable somehow, but again more clicks, more boiler-plate menus (like Service, VPN, etc. now, provided by core).

Not trying to block progress. I've spent a lot of time tweaking these over the years and not satisfied with these overall, but the easy work is already done when we dive into the topic... :)

Cheers,
Franco
